error.js

```diff
@@ -1,75 +1,103 @@
+let factory;
 
-var error = {};
+class AssertionError extends Error {
+  constructor (message, actual, expected) {
+    if (factory == null) factory = require("./");
+    super(message);
+    this.actual = actual;
+    this.expected = expected;
+    this.showDiff = true;
+    Error.captureStackTrace(this, factory);
```

Comment:
after some testing it seems like this works fine. If `factory` is in the call stack it gets trimmed back that far, if its not in the call stack then the stack is unchanged. So if using the validators or notations directly (i.e. not going through the exported factory function), the stacks should still be accurate.